### PR TITLE
Replaced RemoveNotGrantedElementsListener with SecuredManager

### LINKED ISSUE #43: Prevent calling SecuredElementInterface::isAllowed() when there is no token in token storage
[SecuredElementListener](https://github.com/fsi-open/admin-security-bundle/blob/master/EventListener/SecuredElementListener.php#L51) should not call SecuredElementInterface::isAllowed() when there is no token, because it finally throws exception [here](https://github.com/symfony/Security/blob/master/Core/Authorization/AuthorizationChecker.php#L57)

### LINKED ISSUE #73: Do not remove element from manager when it's not allowed for current user
It should be rather omitted during (menu / actions / batch actions) generation than completely removed. Current implementation causes confusion during debugging. Another problematic use case is when list element is allowed but action or batch action should not be allowed. Then additional permission check is required before adding action or batch action to list's datagrid.

### PR DESCRIPTION
**Pull request by szymach**
Resolves #43, #73 and #75

TODO:

- [x] Documentation

### CODE REVIEW

**Comment by szymach** [comment]
@fsi-open

File: CHANGELOG-2.0.md [hunk 1]
```diff
@@ -52,3 +52,19 @@ translation domain. Now they are displayed through flash messages using the defa
 In order for the bundle to work properly, the repository for user class defined
 in the `admin_security_user` parameter needs to implement the `FSi\Bundle\AdminSecurityBundle\Security\User\UserRepositoryInterface`,
 otherwise an exception will be thrown during container compilation.
+
+## Introduced SecuredManager
+
+Instead of removing elements with restricted access from the `admin.manager` service,
+we have created a decorator for it that handles accessing them. This way we do not
+have to rely on the `FSi\Bundle\AdminSecurityBundle\EventListener\RemoveNotGrantedElementsListener`
+and no longer will there be calls to security checks on every kernel request. This
```
Comment (chives) [inline CHANGELOG-2.0.md]: `no longer there will be calls` ?

File: composer.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
     ],
     "require": {
         "php": ">=5.4",
-        "fsi/admin-bundle" : "^2.0@dev",
+        "fsi/admin-bundle" : "^2.1dev,>=2.1.1",
```
Comment (szymach) [inline composer.json]: omg I forgot `@`

**Comment by szymach** [comment]
@JarekW @rn0

**Pull request merged**